### DISCUSSION TITLE
arthas-boot在执行core的时候 为什么是单独开了一个子进程呢

### DISCUSSION THREAD

**Discussion by GayITHub** [alibaba/arthas#2475] [Q&A]
arthas-boot模块中进行了 startArthasClient和startArthasCore这两个模块的动作，
startArthasClient是在当前进程里直接执行的，为什么在startArthasCore的时候要单独启动一个子进程呢？
不是很能想明白这个问题，从实现的角度来考虑，个人觉得和直接在arthas-boot的流程中同步进行arthas-core模块与开启一个子进程去加载arthas-core好像没什么区别，甚至也不用进行输入、输出流的重定向，从流程上来讲更直白？
能不能来个大佬解惑一下

**Comment by hengyunabc** [comment]
因为 arthas-boot 是参考 as.sh 实现的，的确可以简化。
